plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,473 @@
+		records["bf_queue_len"] = *int32Ptr
+	}
+	if int32Ptr, ok := diag.GetBfQueueLenMeanOk(); ok {
+		records["bf_queue_len_mean"] = *int32Ptr
+	}
+	if boolPtr, ok := diag.GetBfActiveOk(); ok {
+		records["bf_active"] = *boolPtr
+	}
+
+	acc.AddFields("slurm_diag", records, tags)
+}
+
+func (s *Slurm) gatherJobsMetrics(acc telegraf.Accumulator, jobs []goslurm.V0038JobResponseProperties) {
+	for i := range jobs {
+		records := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		tags["source"] = s.baseURL.Hostname()
+		if strPtr, ok := jobs[i].GetNameOk(); ok {
+			tags["name"] = *strPtr
+		}
+		if int32Ptr, ok := jobs[i].GetJobIdOk(); ok {
+			tags["job_id"] = strconv.Itoa(int(*int32Ptr))
+		}
+
+		if strPtr, ok := jobs[i].GetJobStateOk(); ok {
+			records["state"] = *strPtr
+		}
+		if strPtr, ok := jobs[i].GetStateReasonOk(); ok {
+			records["state_reason"] = *strPtr
+		}
+		if strPtr, ok := jobs[i].GetPartitionOk(); ok {
+			records["partition"] = *strPtr
+		}
+		if strPtr, ok := jobs[i].GetNodesOk(); ok {
+			records["nodes"] = *strPtr
+		}
+		if int32Ptr, ok := jobs[i].GetNodeCountOk(); ok {
+			records["node_count"] = *int32Ptr
+		}
+		if int64Ptr, ok := jobs[i].GetPriorityOk(); ok {
+			records["priority"] = *int64Ptr
+		}
+		if int32Ptr, ok := jobs[i].GetNiceOk(); ok {
+			records["nice"] = *int32Ptr
+		}
+		if int32Ptr, ok := jobs[i].GetGroupIdOk(); ok {
+			records["group_id"] = *int32Ptr
+		}
+		if strPtr, ok := jobs[i].GetCommandOk(); ok {
+			records["command"] = *strPtr
+		}
+		if strPtr, ok := jobs[i].GetStandardOutputOk(); ok {
+			records["standard_output"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if strPtr, ok := jobs[i].GetStandardErrorOk(); ok {
+			records["standard_error"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if strPtr, ok := jobs[i].GetStandardInputOk(); ok {
+			records["standard_input"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if strPtr, ok := jobs[i].GetCurrentWorkingDirectoryOk(); ok {
+			records["current_working_directory"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if int64Ptr, ok := jobs[i].GetSubmitTimeOk(); ok {
+			records["submit_time"] = *int64Ptr
+		}
+		if int64Ptr, ok := jobs[i].GetStartTimeOk(); ok {
+			records["start_time"] = *int64Ptr
+		}
+		if int32Ptr, ok := jobs[i].GetCpusOk(); ok {
+			records["cpus"] = *int32Ptr
+		}
+		if int32Ptr, ok := jobs[i].GetTasksOk(); ok {
+			records["tasks"] = *int32Ptr
+		}
+		if int64Ptr, ok := jobs[i].GetTimeLimitOk(); ok {
+			records["time_limit"] = *int64Ptr
+		}
+		if strPtr, ok := jobs[i].GetTresReqStrOk(); ok {
+			for k, v := range s.parseTres(*strPtr) {
+				records["tres_"+k] = v
+			}
+		}
+
+		acc.AddFields("slurm_jobs", records, tags)
+	}
+}
+
+func (s *Slurm) gatherNodesMetrics(acc telegraf.Accumulator, nodes []goslurm.V0038Node) {
+	for _, node := range nodes {
+		records := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		tags["source"] = s.baseURL.Hostname()
```

Comment:
See above.

Reply:
I have also preallocated both maps here. Bear in mind that when counting the number of records I only added one for each TRES field. We can't know how many records the'll be parsed into... We can also estimate each TRES field will yield 3 new records and simply increase the hint size passed to `make` by `3 * # of tres fields`.

What's your take?